RESULTS.md

```diff
@@ -570,5 +590,94 @@ examples from the first 400 moves should mostly rid us of those pointless
+even sampling by move would oversample our resign-disabled games. A fixed-number
+sampling would at least guarantee a 10% representation of resign-disabled games.
+
+In terms of bottlenecks, our sampler took a big chunk of time to run, but it
+was still shorter than the training time, so it was not technically a bottleneck
+yet. But we were still training on a single TPU (8 cores) rather than a pod
+(128 cores), so if we were to scale to more TPUs for selfplay, and upgrade our
+trainer to use a pod to compensate, then this shuffler would quickly become the
+bottleneck.
+
+For v14, our selfplay workers would instead directly upload each move to BigTable
+as a (position, next_move, winner) tuple. A set of parallel readers would then
+use BigTable's built-in samplers to read over a specified window of games. In
+order to ensure a 10% ratio of resign-disabled games to resign-enabled games, we
+used the BigTable input readers to enforce a 90:10 ratio.
+
+Another benefit of the BigTable integration was that we could directly run a
+query on BigTable to compute the appropriate resign threshold. Previously, this
+had been done by manually inspecting a Stackdriver graph and editing a flagfile
+on GCS. This meant that the latency of resign threshold updates dropped from
+hours to minutes.
+
+Overall, time between checkpoints dropped from 30 minutes to 20 minutes. (With
+one run being 800 checkpoints, this means that we can complete a run in 11 days.)
+We had much more consistent sampling for resign disabled/enabled games and for
+games of varying lengths. We reduced the latency for resign threshold measurements.
+
+The run was uneventful and was a bit better than previous runs; it escaped random
+play much more quickly, probably because the improved resign threshold latency
+meant that our training data was appropriately pruned of 'easy' positions.
+So all in all, the rewrite didn't create any new pipeline issues.
+
+Around generation 280, we discovered on LCZero forums that AlphaGoZero had used
+init-to-loss for Q values. We switched to init-to-loss from init-to-parent and
+saw our network gain dramatically in strength. Soon, our top model thoroughly
+exceeded our previous best model, and beat LZ's v157, the last model from before
+LZ started mixing in game data from the much stronger ELF.
+
+### v15, clean run with init to loss
+
+This run beat v14 pretty early on, producing a model that was stronger than v14's
+best at generation 330. v15 ended up slightly stronger than v14 but was otherwise
+similar.
+
+Qualitatively, we noticed that V14/V15's models were much sharper,
+meaning that they had more concentrated policy network output and the value
+network output would have stronger opinions on whether it was winning or losing.
+Simultaneously, the calibrated resign threshold for init-to-loss ended up at 0.5
+compared to 0.9 in previous runs. So v14/v15's MCTS was operating in the sweet
```

Comment:
how do you figure this is the sweet spot?

Reply:
we'd previously noticed that MCTS just plays the policy move without reading at all, when it gets close to 0.95